### PR TITLE
Add the ability to specify a regex on an event property

### PR DESCRIPTION
**Pull request by michael-erasmus**
Let's us add regexes to event properties (using the `pattern` nomenclature that Segment uses)

### CODE REVIEW

File: tracking_plan/yaml_event_property.py [hunk 1]
```diff
@@ -31,8 +35,12 @@ def to_json(self):
...
 
-        return {
+        output = {
             'description': self.description,
             'type': p_types,
             'id': f"/properties/properties/properties/{self.name}"
-        }
\ No newline at end of file
+        }
+        if self.pattern:
```
Comment (djfarrelly) [inline tracking_plan/yaml_event_property.py]: Should we only allow the pattern when the `type` is String? From looking on the Protocols UI, they only allow regex w/ String type.

Comment (michael-erasmus) [reply tracking_plan/yaml_event_property.py]: Good call @djfarrelly. What do you think we should do if they type is not `string` but a pattern is present? Should we raise an error or just not supply the pattern?

Comment (djfarrelly) [reply tracking_plan/yaml_event_property.py]: Yeah @michael-erasmus, I think we should raise and error. I think that could be useful as a way to implement validations too:

- If a required yaml key is missing (i.e. `type`) or an invalid value is supplied (a pattern for a non-string), we throw a custom `ValidationError` type
- If an error happens during an `update` - we fail and log the error and event name, property, etc.
- If an error happens during a new `validate` command, we catch the error and log it to the output so one could see all validation errors.

How does that sound?

Comment (djfarrelly) [reply tracking_plan/yaml_event_property.py]: @michael-erasmus, as a follow up, I think if we implement something like that, we can also run it on Travis for branch builds and fail if the validate fails

**Comment by michael-erasmus** [comment]
Hey @djfarrelly!

Thanks for the review and suggestion. I've added some code to check if the event property is of type `string` when a pattern is specified. Let me know how this feels!

**Pull request merged**